compiler/src/dotty/tools/dotc/typer/CrossVersionChecks.scala

```diff
@@ -107,6 +109,18 @@ class CrossVersionChecks extends MiniPhase:
     }
   }
 
+  /** ??? */
+  def checkDeprecatedInheritance(parents: List[Tree])(using Context): Unit = {
+    for parent <- parents
+        psym = parent.tpe.classSymbol
+        if !skipWarning(psym)
+        annot <- psym.getAnnotation(defn.DeprecatedInheritanceAnnot)
+    do
+      val msg = annot.argumentConstantString(0).map(msg => s": $msg").getOrElse("")
+      val since = annot.argumentConstantString(1).map(version => s" (since: $version)").getOrElse("")
```

Comment:
```suggestion
      val since = annot.argumentConstantString(1).filter(!_.isEmpty).map(version => s" (since: $version)").getOrElse("")
```

Reply:
There is no need for the `filter` (See : [tests/warn/i19002.check](https://github.com/lampepfl/dotty/pull/19082/files#diff-7720adf776fa8d0e72ca7c3714722b1eb7db7035dbf7a4cdeaf265b225a60632))

Reply:
In that case, I don't know what dotty is doing with the annotation. It has a default arg that is empty string. I won't press the matter here, but I wonder what happens to the string. In scala 2, we must handle the empty string to avoid `(since: )` text. Sorry I don't already know the answer, I'll check it out later.

Reply:
> It has a default arg that is empty string. I won't press the matter here, but I wonder what happens to the string.

We don't transform annotations in Dotty, therefore, it does not have the knowledge of the default parameter in case we are using named arguments. The only way to have the message `(since: )` is to explicitly write `@deprecatedInheritance(since = "")`.